lib/OpenQA/Schema/Result/JobModules.pm

```diff
@@ -215,10 +182,11 @@ sub split_results($;$) {
     return unless $results; # broken test
     my $schema = OpenQA::Scheduler::schema();
     for my $tm (@{$results->{testmodules}}) {
-        my $r = _insert_tm($schema, $job, $tm);
+        my $r = $job->_insert_tm($schema, $tm);
```

Comment:
That's a nitpick, but I think _ prefixed methods are usually meant as private. Since you edit the function anyway, can you rename it?